src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1237,6 +1237,8 @@ public virtual MethodSymbol TupleUnderlyingMethod
             }
         }
 
+        internal virtual Location ErrorLocation => Locations[0];
```

Comment:
Perhaps `DiagnosticLocation` instead? Also, it looks like this method is only called for `LambdaSymbol`. Consider moving the method to `LambdaSymbol` and making non-virtual for now.